From: Dimitri John Ledkov <xnox@ubuntu.com>
Date: Thu, 26 Jul 2018 15:52:04 +0100
Subject: test-execute: fix execution expectations in containers

Under default LXD/LXC, certain tests fail as shown, thus conditionalize the
expectations... For some cases, the units fail to start due to permission
errors, or do run but exit badly.
---
 src/test/test-execute.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/test/test-execute.c b/src/test/test-execute.c
index fa8efdd..9e2c8e7 100644
--- a/src/test/test-execute.c
+++ b/src/test/test-execute.c
@@ -600,13 +600,19 @@ static void test_exec_privatenetwork(Manager *m) {
 
 static void test_exec_oomscoreadjust(Manager *m) {
         test(m, "exec-oomscoreadjust-positive.service", 0, CLD_EXITED);
-        test(m, "exec-oomscoreadjust-negative.service", 0, CLD_EXITED);
+        if (detect_container() > 0)
+                test(m, "exec-oomscoreadjust-negative.service", 1, CLD_EXITED);
+        else
+                test(m, "exec-oomscoreadjust-negative.service", 0, CLD_EXITED);
 }
 
 static void test_exec_ioschedulingclass(Manager *m) {
         test(m, "exec-ioschedulingclass-none.service", 0, CLD_EXITED);
         test(m, "exec-ioschedulingclass-idle.service", 0, CLD_EXITED);
-        test(m, "exec-ioschedulingclass-realtime.service", 0, CLD_EXITED);
+        if (detect_container() > 0)
+                test(m, "exec-ioschedulingclass-realtime.service", 211, CLD_EXITED);
+        else
+                test(m, "exec-ioschedulingclass-realtime.service", 0, CLD_EXITED);
         test(m, "exec-ioschedulingclass-best-effort.service", 0, CLD_EXITED);
 }
 
